src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -90,12 +90,16 @@ public override async Task<LSP.CompletionItem[]> HandleRequestAsync(LSP.Completi
             var list = await completionService.GetCompletionsAsync(document, position, completionTrigger, options: completionOptions, cancellationToken: cancellationToken).ConfigureAwait(false);
             if (list == null)
             {
-                return Array.Empty<LSP.CompletionItem>();
+                return null;
             }
 
             var lspVSClientCapability = context.ClientCapabilities?.HasVisualStudioLspCapability() == true;
 
-            return list.Items.Select(item => CreateLSPCompletionItem(request, item, lspVSClientCapability, completionTrigger)).ToArray();
+            return new LSP.VSCompletionList
+            {
+                Items = list.Items.Select(item => CreateLSPCompletionItem(request, item, lspVSClientCapability, completionTrigger)).ToArray(),
+                SuggesstionMode = list.SuggestionModeItem != null,
```

Comment:
Not super important, would it be good to file a bug on LSP for the misspelling of this word?

Reply:
Out of curiosity does `SuggestionModeItem` only ever equal not equal `null` at contextually relevant locations? Just trying to make sure it doesn't impact the Razor scenario where you type `@Dat|` and can complete `DateTime` by typing `.`.

Reply:
Based on the comment, it's only non-null when the list should be operating in suggestion mode - http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/Completion/CompletionContext.cs,136

Reply:
for misspelling https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1192571